src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Alert, NativeSyntheticEvent} from 'react-native';
+import {ContextMenuOnPressNativeEvent} from 'react-native-context-menu-view';
+import {useCallback, useContext} from 'react';
+
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import UseMessageHook from '../../../types/hooks/screens/useMessageHook.types';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {Context} from '../../context';
+import {setReplyTarget} from '../../context/actions/chat';
+
+function useMessageHook(): UseMessageHook {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+
+  const setReplyPreview = useCallback((messageItem) => {
+    console.log('Reply Message');
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    console.log('clear Reply target');
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = (messageId: string) => {
+    if (!localDb) return;
+    try {
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+    } catch (error) {
+      console.log('error on delete message:', error);
+    }
+  };
+
+  const onContextMenuPressed = (
+    e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>,
+    messageId: string,
+    message: string
+  ) => {
+    const event = e.nativeEvent;
+    if (event.index === 0) {
+      const messageItem = {
+        username: 'Anonymous User',
+        time: '4h',
+        message: 'sdfgsdMollit esse ea anim exercitation irure quis nostrud est.',
+        messageId: 'sdfs',
+        chatType: '',
+        messageType: 'regular'
+      };
+      setReplyPreview(messageItem);
+    }
+    if (event.index === 1) {
+      ShareUtils.copyTextToClipboard(message);
+    }
+    if (event.index === 2) {
+      Alert.alert('Delete Message?', '', [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Confirm', style: 'destructive', onPress: () => deleteMessage(messageId)}
+      ]);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The hardcoded `messageItem` object on lines 44-50 might not be what you intended. It seems like you would want to create a `messageItem` based on the actual message data passed into the function. Also, consider using a switch statement or mapping object for handling different `event.index` values for better readability and maintainability.

```diff
-       const messageItem = {
-         username: 'Anonymous User',
-         time: '4h',
-         message: 'sdfgsdMollit esse ea anim exercitation irure quis nostrud est.',
-         messageId: 'sdfs',
-         chatType: '',
-         messageType: 'regular'
-       };
+       const messageItem = {
+         username: 'Actual User',
+         time: 'Actual Time',
+         message: message,
+         messageId: messageId,
+         chatType: 'Actual Chat Type',
+         messageType: 'Actual Message Type'
+       };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->